features/netsocket/UDPSocket.h

```diff
@@ -94,17 +94,18 @@ class UDPSocket : public Socket {
     nsapi_size_or_error_t sendto(const SocketAddress &address,
             const void *data, nsapi_size_t size);
 
-    /** Receive a packet over a UDP socket
+    /** Receive a datagram over a UDP socket
      *
-     *  Receives data and stores the source address in address if address
-     *  is not NULL. Returns the number of bytes received into the buffer.
+     *  Receives a datagram and stores the source address in address if address
+     *  is not NULL. Returns the number of bytes written into the buffer. If the
+     *  datagram is larger than the buffer, the excess data is silently discarded.
      *
-     *  By default, recvfrom blocks until data is sent. If socket is set to
-     *  non-blocking or times out, NSAPI_ERROR_WOULD_BLOCK is returned
-     *  immediately.
+     *  By default, recvfrom blocks until a datagram is received. If socket is set to
```

Comment:
Query: Who or what receives the datagram? Please clarify for active voice.

Reply:
How about "blocks until a datagram arrives"?

(POSIX version is much more wordy and specific, with a couple of "if no messages are available..." Don't know if it's worth spelling that out. Could compromise with "has arrived", to imply it may have already arrived before the call.)

Reply:
I like "blocks until a datagram arrives".

Reply:
I don't see much difference between "blocks until a datagram is received" and "blocks until a datagram arrives".

It might even confuse more. Datagram can arrive earlier that user calls this function. Then user receives the datagram from the network stack's buffer. So "datagram arrives" is not totally accurate. Should we clarify this a bit? I found it already clear for me, so I assume SW engineer understand accurately it already.

For a comparison, POSIX standard definition of this same function http://pubs.opengroup.org/onlinepubs/009695399/functions/recvfrom.html


And as this function is `recvfrom()` then the answer to "who receives" is just who ever calls this function.